### PR TITLE
Feature/favorite posts

### LINKED ISSUE #179: As a user, I want to see all the posts I've favorited
Currently there is no use for the favorited tweets.

### PR DESCRIPTION
**Pull request by juanecabellob**
Still working on it, so please don't merge yet. However, comments are welcome! closes #179

### CODE REVIEW

File: posts/templates/posts/_post.html [hunk 1]
```diff
@@ -47,6 +47,9 @@
         {% if request.user.is_authenticated %}
             <a class="mdi-action-grade{% if post.extra.is_favorited %} favorite{% endif %}"
                href="{% url 'posts:favorite' pk=post.original_or_self.pk %}">
+                {% if post.extra.number_of_favorites %}
+                    <span class="badge" style="margin-left: -10px;">{{ post.extra.number_of_favorites }}</span>
```
Comment (Gjum) [inline posts/templates/posts/_post.html:51]: I don't like the [badge](https://youtu.be/EIyixC9NsLI) look.

File: posts/templates/posts/favorite_list.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                    {% with posts=object_list %}
+                        {% for post in posts|dictsortreversed:'created_at' %}
+                            {% if forloop.counter != 1 %}
+                                <hr/>
+                            {% endif %}
+                            {% include 'posts/_post.html' %}
+                        {% empty %}
+                            <br/>
+                            <p class="text-muted lead" style="text-align: center;">
+                                No posts with this hashtag</p>
```
Comment (Gjum) [inline posts/templates/posts/favorite_list.html]: What hashtag?

File: posts/views.py [hunk 3]
```diff
@@ -71,12 +71,14 @@ def set_post_extra(post, request):
     can_be_edited = post.original_or_self().author.pk == request.user.pk
     is_favorited = post.original_or_self().favorites.filter(pk=request.user.pk).exists()
     can_be_deleted = post.author.pk == request.user.pk
+    number_of_favorites = len(post.favorites.all())
```
Comment (Gjum) [inline posts/views.py]: Use a `COUNT` query instead of getting all data from the DB just to get the length.

**Comment by juanecabellob** [comment]
ready to merge. Please comment!

File: templates/base.html [hunk 4]
```diff
@@ -29,6 +29,9 @@
             <div class="navbar-collapse collapse navbar-responsive-collapse">
                 <nav>
                     <ul class="nav navbar-nav">
+                        {% if request.user.is_authenticated %}
+
+                        {% endif %}
```
Comment (metalmatze) [inline templates/base.html]: Ok. That is strange!

File: templates/base.html [hunk 5]
```diff
@@ -54,12 +57,27 @@
...
+<div class="modal fade" id="modal-post-create" tabindex="-1" role="dialog">
+    <div class="modal-dialog" role="document">
+        <div class="modal-content">
+            <div class="modal-body">
+                {% include 'posts/_post_create.html' %}
+            </div>
+        </div>
+    </div>
+</div>
+
```
Comment (metalmatze) [inline templates/base.html:96]: This whole modal thing should be in a different PR because I won't merge it. I clashes with the UI built into search.

File: templates/base.html (hunk 4 above)
Comment (Gjum) [inline templates/base.html]: WTF
